bouncer/tests/broker_fee_collection_test.ts

```diff
@@ -0,0 +1,251 @@
+  );
+
+  // Calculating the fee. Using some strange math here because the SC rounds down on 0.5 instead of up.
+  const divisor = BigInt(1 / (testCommissionBps / 10000));
+  const expectedIncrease =
+    depositAmount / divisor + (depositAmount % divisor > divisor / 2n ? 1n : 0n);
+  assert.strictEqual(
+    increase,
+    expectedIncrease,
+    `Unexpected increase in the ${asset} earned broker fees. Did the broker commission change?`,
+  );
+
+  // Withdraw the broker fees
+  const withdrawalAddress = await newAddress(asset, seed ?? randomBytes(32).toString('hex'));
+  const useableWithdrawalAddress =
+    asset === Assets.DOT ? decodeDotAddressForContract(withdrawalAddress) : withdrawalAddress;
+  const chain = chainShortNameFromAsset(asset);
+  console.log(`${chain} withdrawalAddress:`, withdrawalAddress);
+  const balanceBeforeWithdrawal = await getBalance(asset, withdrawalAddress);
+  console.log(
+    `Withdrawing broker fees to ${useableWithdrawalAddress}, balance before: ${balanceBeforeWithdrawal}`,
+  );
+  const observeWithdrawalRequested = observeEvent(
+    'swapping:WithdrawalRequested',
+    chainflip,
+    (event) =>
+      event.data.destinationAddress[chain]?.toLowerCase() ===
+      useableWithdrawalAddress.toLowerCase(),
+  );
+
+  // Only allow one withdrawal at a time to stop nonce issues
+  while (withdrawLockout) {
+    await sleep(100);
+  }
+  withdrawLockout = true;
+  const withdrawal = await chainflip.tx.swapping
+    .withdraw(asset, {
+      [chain]: useableWithdrawalAddress,
+    })
+    .signAndSend(broker);
+
+  console.log('Submitted Withdrawal:', JSON.stringify(withdrawal));
+  const withdrawalRequestedEvent = await observeWithdrawalRequested;
+  console.log(`Withdrawal requested, egressId: ${withdrawalRequestedEvent.data.egressId}`);
+  const BatchBroadcastRequestedEvent = await observeEvent(
+    ':BatchBroadcastRequested',
+    chainflip,
+    (event) =>
+      event.data.egressIds.some(
+        (egressId: EgressId) =>
+          egressId[0] === withdrawalRequestedEvent.data.egressId[0] &&
+          egressId[1] === withdrawalRequestedEvent.data.egressId[1],
+      ),
+  );
+  console.log(
+    `Batch broadcast requested, broadcastId: ${BatchBroadcastRequestedEvent.data.broadcastId}`,
+  );
+
+  withdrawLockout = false;
+
+  await observeBalanceIncrease(asset, withdrawalAddress, balanceBeforeWithdrawal);
+
+  // Check that the balance after withdrawal is correct after deducting withdrawal fee
+  const balanceAfterWithdrawal = await getBalance(asset, withdrawalAddress);
+  console.log(`${asset} Balance after withdrawal:`, balanceAfterWithdrawal);
+  const balanceAfterWithdrawalBigInt = BigInt(
+    amountToFineAmount(balanceAfterWithdrawal, assetDecimals[asset]),
+  );
+  const expectedBalanceAfterWithdrawal =
+    BigInt(amountToFineAmount(balanceBeforeWithdrawal, assetDecimals[asset])) +
+    earnedBrokerFeesAfter -
+    maxWithdrawalFee[asset];
+  const detectWithdrawalGasFee = -(
```

Comment:
Would be more natural to write `balanceBefore + earnedFees - balanceAfter`, then you won't need to negate it.